bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -66,9 +82,45 @@ public void propertyChange(PropertyChangeEvent evt) {
 			int percentVisible = (int) (100 * a.getMean(Circumstance.TIMING_WINDOW_OPEN, false));
 			switch (percentVisible) {
 			case 100: // good!
-				if (a.getObs().getTimingWindows().size() > 0)
+				if (a.getObs().getTimingWindows().size() > 0 && night != null)
 					mm.addMarker(true, this, Severity.Info, "Timing constraint is met.", v, a);
-				break;
+
+                    // Report timing windows
+                    final TimingWindowSolver timingWindowSolver = new TimingWindowSolver(a.getObs());
+                    final Iterator<Interval> timingWindowIntervals = timingWindowSolver.solve(night).iterator();
```

Comment:
If `percentVisible` is `100` here, what are we doing again?  The idea is to report on other times besides the current `Alloc` where we could also place this observation?

Reply:
The idea, if I understood correctly, is that if all constraints are met we give a notice message showing where they are so the user can rearrange allocations without having to find out manually and writing down the constraint ranges.

If any constraints is not met then we get an error right away so we don't need to show the constraints info, it's already on the error. The timing window error is kind of weird because anything that is not `100` visible is already an error, but the error reported is an overall percentage of all allocations, not of the specific allocation not meeting the constraint.